transports/transports.go

```diff
@@ -77,7 +77,9 @@ func ListNames() []string {
 	defer kt.mu.Unlock()
 	var names []string
 	for _, transport := range kt.transports {
-		names = append(names, transport.Name())
+		if !transport.Deprecated() {
```

Comment:
It does feel a bit weird for a transport to declare _itself_ deprecated (deprecated for what purpose? etc.); though with the `transports` package intentionally not knowing all other transports, the alternatives are also ugly.

I guess I would mildly prefer to keep this information inside the `transports` package (I guess a set of strings as transport names?), not in the individual transports themselves.

---

Separately, the `ListNames` method’s documentation needs to be updated to say that it is not exhaustive / does not show deprecated transports, and that the purpose is to offer recommended alternatives in an UI.  [Either way, a human needs to read the transport’s documentation to figure out what name to use within the transport, so this is unsuitable for automated processing.]